src/etc/rust_analyzer_eglot.el

```diff
@@ -2,28 +2,31 @@
   .((eglot-workspace-configuration
      . (:rust-analyzer
         ( :check ( :invocationLocation "root"
-                   :invocationStrategy "once"
-                   :overrideCommand ["python3"
-                                     "x.py"
-                                     "check"
-                                     "--json-output"])
-          :linkedProjects ["Cargo.toml"
-                           "src/tools/x/Cargo.toml"
-                           "src/bootstrap/Cargo.toml"
-                           "src/tools/rust-analyzer/Cargo.toml"
-                           "compiler/rustc_codegen_cranelift/Cargo.toml"
-                           "compiler/rustc_codegen_gcc/Cargo.toml"]
-          :rustfmt ( :overrideCommand ["build/host/rustfmt/bin/rustfmt"
-                                       "--edition=2021"])
-          :procMacro ( :server "build/host/stage0/libexec/rust-analyzer-proc-macro-srv"
-                       :enable t)
-          :cargo ( :buildScripts ( :enable t
-                                   :invocationLocation "root"
-                                   :invocationStrategy "once"
-                                   :overrideCommand ["python3"
-                                                     "x.py"
-                                                     "check"
-                                                     "--json-output"])
-                   :sysrootSrc "./library"
-                   :extraEnv (:RUSTC_BOOTSTRAP "1"))
-          :rustc ( :source "./Cargo.toml" )))))))
+                                       :invocationStrategy "once"
+                                       :overrideCommand ["python3"
+                                                         "x.py"
+                                                         "check"
+                                                         "--json-output"])
+                 :linkedProjects ["Cargo.toml"
+                                  "src/tools/x/Cargo.toml"
+                                  "src/bootstrap/Cargo.toml"
+                                  "src/tools/rust-analyzer/Cargo.toml"
+                                  "compiler/rustc_codegen_cranelift/Cargo.toml"
+                                  "compiler/rustc_codegen_gcc/Cargo.toml"]
+                 :rustfmt ( :overrideCommand [(concat (project-root (eglot--current-project))
+                                                      "build/host/rustfmt/bin/rustfmt")
```

Comment:
Ah well, slight issue here. Because rust-analyzer doesn't support something like invocationLocation for rustfmt, it needs an absolute path to be provided here otherwise it'll look for the closest Cargo.toml and try to call e.g. src/bootstrap/build/host/rustfmt/bin/rustfmt which obviously does not exist.

The vscode config gets around this by sticking ${workspaceFolder} in the string but that doesn't work here because it's a vscode thing. I tried getting around it with this concat trick, which does make rustfmt work, but then other parts of LSP choke on the input while trying to parse it because Emacs is not expecting there to be any functions getting called in this file. Not great.

The optimal solution would be to fix rust-analyzer at some point, what workaround should we go with for now? The easiest one would be to leave a note about how the user needs to adjust the path by hand. We could automate it as well where we just fix the path in setup.rs before the file is written to its proper location, but then that'll break the hash-based versioning because now what's actually used would always be different from the template.

Reply:
For now I left the path as `build/host/rustfmt/bin/rustfmt`, this will make it work sometimes but break at other times depending on what rust-analyzer decides its root to be. At least when it does break it's just the formatting and it doesn't take the whole LSP along with it.